src/hooks/useModal/index.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { useState, useCallback, ReactElement, createContext, useMemo, useContext } from 'react';
+import { createPortal } from 'react-dom';
+import { match } from 'ts-pattern';
+
+import { noop } from '../../utils/utils';
+import Modal, { type ModalProps } from '../../ui/Modal';
+import { MODAL_ROOT } from './ModalRoot';
+
+export type OpenGlobalModalProps = {
+  modalProps: ModalProps;
+  childElement: ReactElement;
+};
+export interface GlobalModalProviderProps {
+  children: ReactElement;
+}
+export interface GlobalModalContextInterface {
+  openModal: (props: OpenGlobalModalProps) => void;
+}
+
+const GlobalModalContext = createContext<GlobalModalContextInterface>({
+  openModal: noop,
+});
+
+export const GlobalModalProvider = ({ children }: GlobalModalProviderProps) => {
+  // Idea from https://dev.to/nurislamov/simple-modals-queue-in-react-4g6c
+  const [globalModalQueue, setGlobalModalQueue] = useState<Array<OpenGlobalModalProps>>([]);
```

Comment:
Which is better?
```
useState<OpenGlobalModalProps[]>
```

Reply:
am... we decided to not use this pattern in this project, but it's long time ago
```
type[]
```